src/components/gridContainer.js

```diff
@@ -154,28 +155,30 @@ class GridContainer extends Component {
     }
    
     saveGrid() {
-        if (this.state.isDirty) {
-            // call parse endpoint first then save
-            const prom = this.postPreviewData(this.processHandsontableData());
-            prom.then((previewData) => { 
+        if (this.tableHasAtLeastOneHeaderRowOrColumn() && this.tableHasAppropriateHeaderColumnsSet() && this.tableHasAppropriateHeaderRowsSet()) {
```

Comment:
Could this conditional be extracted to a method like `isValid`? Or `validate` as it also sets the error as well as setting the error message.